src/Play.ts

```diff
@@ -0,0 +1,573 @@
+import { Bonus, Game, Letter, PlayerIndex, SparseArray } from "./types";
+import words from "./words.json";
+import {
+  canReachPlayer,
+  horizontalSlice,
+  isValidMatch,
+  sliceToRegex,
+  verticalSlice,
+} from "./utils/gameBoardHelper";
+import { tileValues } from "./constants";
+
+let lastAwaitTime = 0;
+
+async function waitEventLoop() {
+  let rv = null;
+
+  if (Date.now() - lastAwaitTime > 5) {
+    rv = new Promise((resolve) => setTimeout(resolve));
+  }
+
+  lastAwaitTime = Date.now();
```

Comment:
This should be inside the if statement I think (otherwise any series of blocking code that each blocks for less than 5ms will collectively block for longer). Might need to up the statement from checking 5 ms to something a bit more if that's moved to right place.